ledger/participant-state/src/main/scala/com/daml/ledger/participant/state/v1/Update.scala

```diff
@@ -107,17 +144,85 @@ object Update {
     * @param recordTime
     *   The ledger-provided timestamp at which the package was uploaded.
     *
+    * @param submissionId
+    *   submissionId of the package upload command.
+    *
     */
   final case class PublicPackageUploaded(
       archive: DamlLf.Archive,
       sourceDescription: Option[String],
       participantId: ParticipantId,
-      recordTime: Timestamp)
+      recordTime: Timestamp,
+      submissionId: SubmissionId)
       extends Update {
     override def description: String =
       s"""Public package uploaded: ${archive.getHash}"""
   }
 
+  /** Signal that the package upload entry has been Accepted.
+    *
+    * Initially this will be visible to all participants in the current open world,
+    * with a possible need to revisit as part of the per-party package visibility work
+    * https://github.com/digital-asset/daml/issues/311.
+    *
+    * @param participantId
+    *   The participant through which the package was uploaded. This field
+    *   is informative, and can be used by applications to display information
+    *   about the origin of the package.
+    *
+    * @param recordTime
+    *   The ledger-provided timestamp at which the package was uploaded.
+    *
+    * @param submissionId
+    *   submissionId of the package upload command.
+    *
+    * Consider whether an enumerated set of reject reasons a la [[RejectionReason]] would be helpful, and whether the same breadth of reject
+    * types needs to be handled for package upload entry rejects
+    */
+  final case class PackageUploadAccepted(
```

Comment:
@mziolekda: can we go through the reasoning again why `PublicPackageUploaded` and `PackageUploadAccepted` are separate messages?

Reply:
PublicPackageUploaded is per package (if there is a dar with multiple packages, there will be three PublicPackageUploaded messages)

PackageUploadAccepted is for the entire submission (so for all 3 packages in the example)

If packages have already been uploaded (it is a duplicate upload), there will be no PublicPackageUploaded message and there will be a single PackageUploadAccepted message

If there are multiple packages and any of them is invalid or rejected, there will be no PublicPackageUploaded messages and there will be a single PackageUploadRejected message